src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,73 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { FilterTodos } from '../../types/FilterTodos';
+
+type Props = {
+  query: string;
+  filterParameter: FilterTodos | string;
+  onInputChange: (str: string) => void;
+  onSelectChange: (field: string) => void;
+};
+
+export const TodoFilter: React.FC<Props> = (
+  {
+    query,
+    filterParameter,
+    onInputChange,
+    onSelectChange,
+  },
+) => {
+  const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onInputChange(event.target.value);
+  };
+
+  const handleSelect
+  = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    return onSelectChange(event.target.value);
+  };
+
+  const clear = () => onInputChange('');
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={filterParameter}
+            onChange={handleSelect}
+          >
+            <option value={FilterTodos.All}>All</option>
+            <option value={FilterTodos.Active}>Active</option>
+            <option value={FilterTodos.Completed}>Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={handleInput}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        {query && (
+          <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
```

Comment:
Try not to disable eslint and figure out the problem. In this case you have to add [aria-label ](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label)attribute